dts/bindings/serial/silabs,gecko-usart.yaml

```diff
@@ -21,20 +21,16 @@ properties:
 
     location-rx:
       type: array
-      required: true
       description: RX pin configuration defined as <location port pin>
```

Comment:
note: these properties should actually be flagged as deprecated and removed when pinmux (and any legacy pin config methods) are deleted from tree.

Reply:
Marked location properties as deprecated.